src/lib/rpcWallet.js

```diff
@@ -107,31 +105,20 @@ export default {
       this.sdk = null;
     }
   },
-  getAeppOrigin(aepp) {
-    const {
-      connection: {
-        port: {
-          sender: { url },
-        },
-      },
-    } = aepp;
-    return extractHostName(url);
+  getAeppHost(aepp) {
+    return new URL(aepp.connection.port.sender.url).host;
   },
   async shouldOpenPopup(aepp, action) {
```

Comment:
Every method like `onSign` calls firstly `shouldOpenPopup` (wich calls `checkPermissions` from permissions module) and then calls `checkAeppPermissions` (wich calls `getAeppAccountPermission` from permissions module). Can we combine `shouldOpenPopup` and `checkAeppPermissions` into a single function extracted to permissions module?

Reply:
The general idea: if we created a permissions module let's move all related stuff to there.